projects/004-exercise-tracker/jimmyhu/src/api/users/usersapi.ts

```diff
@@ -0,0 +1,125 @@
+const exps: any = require('express')
+
+const router = exps.Router()
+
+
+// global variables for store the information
+let users: user[] = []
+let exercises: exercise[] = []
+let invalidID:  number[] = []
+
+
+// user interface, the Id will be a random number for now
+interface user {
+    'username' : string,
+    '_id': number
+}
+
+// exercise extend the user interface due the structure require the same information anyway
+interface exercise extends user {
+    "description": string,
+    "duration": number,
+    "date": Date|string,
+}
+
+// function for check if the date is in the interval
+function checkdate (Indate:string|Date, start: string, end: string): boolean{
+    if (new Date(Indate) > new Date(start) && new Date(Indate) < new Date(end)){
+        return true
+    } else if (start == undefined || end == undefined){
+        return true
+    }
+    return false
+}
+
+
+// base post request, the id is a random number auto-generated, for avoid double id there are a check on a list
+router.post('/', (req: any, res:any): void => {
+    let username: string = req.body.name
+    let randID: number = Math.floor(Math.random() * 1000)
+    while (invalidID.includes(randID)){
```

Comment:
Per invalidID intendi ID che sono stati già utilizzati in precedenza immagino? Se si, potresti usare un nome più esplicativo, tipo alredyUsedIds, così si capisce subito cosa sono.

In ogni caso, potresti generare id unici usando uuid, così non avrai bisogno di verificarne l'unicità

Reply:
Corretto, Grazie per il consiglio